lib/spack/spack/test/cmd/create.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import argparse
+import os
+import pytest
+
+import spack.cmd.create
+import spack.util.editor
+
+from spack.main import SpackCommand
+
+
+create = SpackCommand('create')
+
+
+@pytest.fixture("module")
+def cmd_create_repo(tmpdir_factory):
+    repo_namespace = 'cmd_create_repo'
+    repodir = tmpdir_factory.mktemp(repo_namespace)
+    repodir.ensure(spack.repo.packages_dir_name, dir=True)
+    yaml = repodir.join('repo.yaml')
+    yaml.write("""
+repo:
+    namespace: cmd_create_repo
+""")
+
+    db = spack.repo.RepoPath(str(repodir))
+    with spack.repo.swap(db):
+        yield repodir
+
+    # shutil.rmtree(str(repodir))
+
+
+@pytest.fixture(scope='module')
+def parser():
+    """Returns the parser for the module"""
+    prs = argparse.ArgumentParser()
+    spack.cmd.create.setup_parser(prs)
+    return prs
+
+
+@pytest.fixture
+def mock_editor(monkeypatch):
+    def _editor(*args, **kwargs):
+        return
+
+    monkeypatch.setattr(spack.util.editor, 'editor', _editor)
+    yield
+
+
+def test_create_template(parser, cmd_create_repo):
+    """Test template creation."""
+    repodir = cmd_create_repo
+
+    name = 'test-package'
+    args = parser.parse_args(['--skip-editor', name])
+    spack.cmd.create.create(parser, args)
+
+    filename = str(repodir.join(spack.repo.packages_dir_name, name,
```

Comment:
I think if `repo.get` can retrieve the package that would be sufficient: as it is, this test reaches into some of the details that are best left to `repo` itself.

Reply:
Not sure I completely follow.  So it's not work making sure the test package inherits from Package and has an install method when creating a basic package template?

Reply:
I think that's useful but IIRC you can retrieve a path to a module if you obtain the module. So if you use `repo.get` to get the associated package module you can extract the path that way. The main thing I think is worth avoiding here is a dependence on knowing how `repo` constructs paths to its packages.

Reply:
Ah!  Now I understand your issue.  Thanks.